includes/class-wc-fgc-update-variation-cart.php

```diff
@@ -0,0 +1,387 @@
+<?php
+/**
+ * Update Variations in cart.
+ *
+ * Handles all Variation Update related functionalities.
+ *
+ *
+ * @since 1.0.0
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+if ( class_exists( 'WC_FGC_Update_Variation_Cart' ) ) {
+	return;
+}
+
+/**
+ * Main WC_FGC_Update_Variation_Cart Class
+ *
+ * @class WC_FGC_Update_Variation_Cart
+ * @package Class
+ * @author   Kevin
+ * @version 1.0.0
+ */
+class WC_FGC_Update_Variation_Cart {
+	/**
+	 * The plugin version
+	 *
+	 * @var string
+	 */
+	public static $version = '1.0.0';
+	/**
+	 * The plugin name
+	 *
+	 * @var string
+	 */
+	public static $name = 'wc_fgc';
+
+	/**
+	 * WC Update Variation Cart constructor
+	 *
+	 * @since 1.0.0
+	 */
+	public static function init() {
+
+		// Enqueue required js and css.
+		add_filter( 'wp_enqueue_scripts', array( __CLASS__, 'enqueue_cart_script' ) );
+
+		// Add edit option on the cart page.
+		add_action( 'woocommerce_after_cart_item_name', array( __CLASS__, 'variation_update_data' ), 1, 2 );
+
+		// Handle the ajax request of the update cart.
+		add_action( 'wp_ajax_wc_fgc_get_product_html', array( __CLASS__, 'get_variation_html' ) );
+		add_action( 'wp_ajax_nopriv_wc_fgc_get_product_html', array( __CLASS__, 'get_variation_html' ) );
+
+		// Update the cart as per the user choice.
+		add_action( 'wp_ajax_wc_fgc_update_variation_in_cart', array( __CLASS__, 'update_variation_in_cart' ) );
+		add_action( 'wp_ajax_nopriv_wc_fgc_update_variation_in_cart', array( __CLASS__, 'update_variation_in_cart' ) );
+
+		// Show error message on the checkout page.
+		add_action( 'woocommerce_before_checkout_process', array( __CLASS__, 'check_is_valid_product' ) );
+	}
+
+	/**
+	 * Enqueue needed js and css
+	 *
+	 * @name enqueue_cart_script
+	 * @param string $page   Name of the page.
+	 * @since 1.0.0
+	 */
+	public static function enqueue_cart_script( $page ) {
+
+		// check is cart page or not.
```

Comment:
slightly pedantic, but don't forget that all comments need to start with an uppercase and end with a period. may as well get it here as eventually this will go in core and eventually we'll add the code standards rules checking to FGC.

Reply:
True, i guess i'll refactor everything.